### PR TITLE
Add snackbar message on empty folder drop

### PR DESCRIPTION
**Pull request by jomunker**
This PR adds a snackbar message, when an empty folder is dropped in the DAM. 

## Screen recording
https://github.com/vivid-planet/comet/assets/56400587/edc3ed28-d057-4526-a3e8-069881d76914

### CODE REVIEW

**Comment by jomunker** [comment]
Currently react-dropzone can't detect empty folders natively and only indirectly if a drop has been registered but there are no accepted and rejected files. Therefore if an empty folder is dropped with something else it won't detect it.

File: packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx [hunk 1]
```diff
@@ -243,6 +246,15 @@ const FolderDataGrid = ({
         setHoveredId(null);
     };
 
+    const getSnackbarElement = (message: string) => (
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx:249]: Why did you create this function? It is only used in one place...

Comment (jomunker) [reply packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx:249]: I had more different messages in the beginning but then realized I don't need them. This is a relict so I can remove it if you want :)

**Review by thomasdax98 — CHANGES_REQUESTED** [review]

**Review by thomasdax98 — APPROVED** [review]

File: packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx [hunk 2]
```diff
@@ -264,10 +276,17 @@ const FolderDataGrid = ({
...
+        onDrop: async (acceptedFiles: File[], fileRejections: FileRejection[], event) => {
             hideHoverStyles();
             hideUploadFooter();
+
             await fileUploadApi.uploadFiles({ acceptedFiles, fileRejections }, currentFolderId);
+
+            // react-dropzone doesn't support folder drops natively
+            // the only way to find out if an empty folder was dropped is if there are no rejected files and no accepted files
+            if (!fileRejections.length && !acceptedFiles.length) {
+                snackbarApi.showSnackbar(emptyFolderSnackbarElement);
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx:288]: Please "create" the element directly here:

```tsx
snackbarApi.showSnackbar(
<Snackbar
  anchorOrigin={{ vertical: "bottom", horizontal: "left" }}
  autoHideDuration={5000}
  TransitionComponent={(props: SlideProps) => <Slide {...props} direction="right" />}
  message={<FormattedMessage id="comet.dam.upload.noEmptyFolders" defaultMessage={"Empty folders can't be uploaded"} />}
/>
)
```

Comment (johnnyomair) [reply packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx:288]: Why is the slight in direction from the right? Should we change this for all Snackbars?

Comment (jomunker) [reply packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx:288]: Why create the element there? It makes it less readable imo 🤔

Comment (johnnyomair) [reply packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx:288]: > Why create the element there? It makes it less readable imo 🤔

I prefer to create elements as close to where they are used. When reading the code I don't have to "jump" to another place to understand it. But you're right, this is an opinionated suggestion. If you prefer it the other way, we leave it like that. 👍🏼

> I just copied the snackbar element from somewhere else. I can create a custom snackbar element to streamline the usage if you want

We'll discuss this with UX and change it then. You may leave it for now.

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**